lib/doorkeeper/oauth/assertion_access_token_request.rb

```diff
@@ -39,114 +39,119 @@ def authorize
-        # >
-        # >   assertion
-        # >         REQUIRED.  The assertion.
-        # >
-        # > - draft-ietf-oauth-v2-10
-        # @see http://tools.ietf.org/html/draft-ietf-oauth-v2-10#section-4.1.3
-        #
-        # Newer versions of ietf-oauth-v2 don't need assertion_type. So it's still optional.
-        def validate_assertion
-          assertion, assertion_type = server.parameters.values_at(:assertion, :assertion_type)
+      ##
+      # >   The value of the "grant_type" is "urn:ietf:params:oauth:grant-
+      # >   type:jwt-bearer".
+      # >   The value of the "assertion" parameter MUST contain a single JWT.
+      # > - draft-ietf-oauth-jwt-bearer-12
+      #
+      # @see https://tools.ietf.org/html/draft-ietf-oauth-jwt-bearer-12#section-2.1
+      #
+      # >   assertion_type
+      # >         REQUIRED.  The format of the assertion as defined by the
+      # >         authorization server.  The value MUST be an absolute URI.
+      # >
+      # >   assertion
+      # >         REQUIRED.  The assertion.
+      # >
+      # > - draft-ietf-oauth-v2-10
+      # @see http://tools.ietf.org/html/draft-ietf-oauth-v2-10#section-4.1.3
+      #
+      # Newer versions of ietf-oauth-v2 don't need assertion_type. So it's still optional.
+      def validate_assertion
+        assertion, assertion_type = server.parameters.values_at(:assertion, :assertion_type)
 
-          if assertion_type and assertion_type != 'urn:ietf:params:oauth:grant-type:jwt-bearer'
-            raise StandardError.new('Assertion type not valid. Expected urn:ietf:params:oauth:grant-type:jwt-bearer')
-          end
+        if assertion_type and assertion_type != 'urn:ietf:params:oauth:grant-type:jwt-bearer'
+          raise StandardError.new('Assertion type not valid. Expected urn:ietf:params:oauth:grant-type:jwt-bearer')
+        end
 
-          if configuration.jwt_use_application_public_key_as_key
-            payload, header = JWT.decode(assertion, nil, false)
-            raise StandardError.new('App ID is not correct') unless OAuth::Client.find(payload['iss']).present?
-            if payload['iss'] =~ /^es_k_/
-              api_secret = OAuth::Client.find(payload['iss']).application.secret
-              payload, header = JWT.decode(assertion, api_secret, true, { algorithm: header['alg'] })
-            else
-              public_key =  OAuth::Client.find(payload['iss']).application.public_key
-              payload, header = JWT.decode(assertion, OpenSSL::PKey::RSA.new(public_key), true, { algorithm: header['alg'] })
-            end
-          else
-            payload, header = JWT.decode(assertion, configuration.jwt_key)
+        if configuration.jwt_use_application_public_key_as_key
+          payload, header = JWT.decode(assertion, nil, false)
+          raise StandardError.new('App ID is not correct') unless OAuth::Client.find(payload['iss']).present?
+          case header['alg']
```

Comment:
is this needed if we don't update the key? do you think that one day we will give the key again, not only the token? to do oauth maybe?